geomesa-hbase/geomesa-hbase-spark/src/main/scala/org/locationtech/geomesa/spark/hbase/HBaseSpatialRDDProvider.scala

```diff
@@ -37,6 +40,14 @@ class HBaseSpatialRDDProvider extends SpatialRDDProvider {
 
     val ds = DataStoreConnector[HBaseDataStore](dsParams)
 
+    if (User.isSecurityEnabled) {
+      // configure auth token so that the HBase TokenUtil can pick it up for
+      // authentication in the remote worker processes - the keytab won't be accessible there
+      // needs to be in UserGroupInformation.getCurrentUser().getCredentials()
+      val token = TokenUtil.obtainToken(ds.connection, User.getCurrent)
+      UserGroupInformation.getCurrentUser.getCredentials.addToken(token.getService, token)
```

Comment:
What does setting this token do for us?  

Is it getting added to the Hadoop configuration object?

Reply:
it doesn't get added to the configuration, but it will get added the to hadoop Job (as hadoop keeps credentials there instead of the conf for whatever reason). there is some code in spark that looks in `UserGroupInformation.getCurrentUser().getCredentials()` for hbase delegation tokens, as that's where hbase will set the tokens in their input formats. i was hoping it would just work without us doing anything, but for some reason it wasn't added unless i did so explicitly.

Reply:
the HBase TokenUtil referenced in the comment is the class that gets invoked and expects the token to be there.